drivers/SmartThings/harman-luxury/src/init.lua

```diff
@@ -0,0 +1,390 @@
+        end
+      end
+      -- check for a audio track data change
+      if changes["audioTrackData"] then
+        local audioTrackData = changes["audioTrackData"]
+        local trackdata = {}
+        if type(audioTrackData.title) == "string" then
+          trackdata.title = audioTrackData.title
+        end
+        if type(audioTrackData.artist) == "string" then
+          trackdata.artist = audioTrackData.artist
+        end
+        if type(audioTrackData.album) == "string" then
+          trackdata.album = audioTrackData.album
+        end
+        if type(audioTrackData.albumArtUrl) == "string" then
+          trackdata.albumArtUrl = audioTrackData.albumArtUrl
+        end
+        if type(audioTrackData.mediaSource) == "string" then
+          trackdata.mediaSource = audioTrackData.mediaSource
+        end
+        -- if track changed
+        device:emit_event(capabilities.audioTrackData.audioTrackData(trackdata))
+
+        device:emit_event(capabilities.mediaPlayback.supportedPlaybackCommands(
+          audioTrackData.supportedPlaybackCommands) or { "play", "stop", "pause" })
+        device:emit_event(capabilities.mediaTrackControl.supportedTrackControlCommands(
+          audioTrackData.supportedTrackControlCommands) or { "nextTrack", "previousTrack" })
+        device:emit_event(capabilities.audioTrackData.totalTime(audioTrackData.totalTime or 0))
+      end
+      -- check for a media presets change
+      if changes["mediaPresets"] and type(changes["mediaPresets"]) == "table" then
+        device:emit_event(capabilities.mediaPresets.presets(changes["mediaPresets"]))
+      end
+      -- check for a media input source change
+      if changes["mediaInputSource"] then
+        device:emit_event(capabilities.mediaInputSource.inputSource(changes["mediaInputSource"]))
+      end
+      -- check for a volume value change
+      if changes["volume"] then
+        device:emit_event(capabilities.audioVolume.volume(changes["volume"]))
+      end
+      -- check for a mute value change
+      if changes["mute"] ~= nil then
+        if changes["mute"] then
+          device:emit_event(capabilities.audioMute.mute.muted())
+        else
+          device:emit_event(capabilities.audioMute.mute.unmuted())
+        end
+      end
+    end
+  end
+end
+
+local function create_check_for_updates_thread(device)
+  local old_timer = device:get_field(const.UPDATE_TIMER)
+  if old_timer ~= nil then
+    log.info(string.format("create_check_for_updates_thread: dni=%s, remove old timer", device.device_network_id))
+    device.thread:cancel_timer(old_timer)
+  end
+
+  log.info(string.format("create_check_for_updates_thread: dni=%s", device.device_network_id))
+  local new_timer = device.thread:call_on_schedule(const.UPDATE_INTERVAL, function()
+    check_for_updates(device)
+  end, "value_updates_timer")
+  device:set_field(const.UPDATE_TIMER, new_timer)
+end
+
+local function create_check_health_thread(device)
```

Comment:
There is non-trivial overhead to creating many tasks; in the case where a user has many speakers in their home, there could be a lot of overhead with having a task per device. I would recommend creating a single background task that monitors for IP address changes and then fans out to the appropriate device.